shared/src/main/java/me/neznamy/tab/shared/proxy/ProxyTabPlayer.java

```diff
@@ -141,7 +141,7 @@ public boolean hasInvisibilityPotion() {
     public boolean hasPermission(@NotNull String permission) {
         if (TAB.getInstance().getConfiguration().isBukkitPermissions()) {
             sendPluginMessage(new PermissionRequest(permission));
-            return permissions != null && permissions.getOrDefault(permission, false);
+            return permissions.getOrDefault(permission, false);
```

Comment:
This will throw NPE if assigning groups by permissions & bukkit permission manager is enabled, because permissions will be checked in TabPlayer constructor, before this map was assigned in ProxyTabPlayer constructor.

Reply:
most likely I was wrong